sstable_generation_loading_test.py

```diff
@@ -27,45 +28,68 @@ def fixture_add_additional_log_patterns(self, fixture_dtest_setup):
         fixture_dtest_setup.allow_log_errors = True
 
     upgrade_from = None
-    compact = False
+    test_compact = False
+
+    def compact(self):
+        return self.fixture_dtest_setup.cluster.version() < MAJOR_VERSION_4 and self.test_compact
 
     def create_schema(self, session, ks, compression):
         create_ks(session, ks, rf=2)
-        create_cf(session, "standard1", compression=compression, compact_storage=self.compact)
+        create_cf(session, "standard1", compression=compression, compact_storage=self.compact())
         create_cf(session, "counter1", compression=compression, columns={'v': 'counter'},
-                  compact_storage=self.compact)
+                  compact_storage=self.compact())
 
     def test_sstableloader_compression_none_to_none(self):
+        if self.__class__.__name__ != 'TestBasedSSTableLoader' and self.upgrade_from is None:
```

Comment:
Since these two lines are redundant. Perhaps refactor this into a small method and reuse?

Reply:
Fixed by adding skip_base_class_test which calls pytest.skip with an explanation.